data/src/main/java/com/microsoft/azure/kusto/data/exceptions/KustoClientException.java

```diff
@@ -0,0 +1,13 @@
+package com.microsoft.azure.kusto.data.exceptions;
+
+public abstract class KustoClientException extends Exception {
+    public KustoClientException(String message) {
+        super(message);
+    }
+
+    public KustoClientException(String message, Exception exception) {
+        super(message, exception);
+    }
+
+    public abstract TriState isPermanent();
```

Comment:
I think permanence is conceptually a property of this class ("permanent"), as each instance has its own value.

I would therefore make this class concrete instead  of abstract, include a property called "permanent", and this would be a concrete implementation of isPermanent() and simply return the value of the property.

Reply:
sounds good